python_modules/dagster-externals/dagster_externals/io/base.py

```diff
@@ -0,0 +1,57 @@
+import importlib
+import json
+from abc import ABC, abstractmethod
+from typing import cast
+
+from .._protocol import (
+    DAGSTER_EXTERNALS_ENV_KEYS,
+    ExternalExecutionContextData,
+    IOSpec,
+    Notification,
+)
+from .._util import DagsterExternalError, param_from_env
+
+
+class ExternalExecutionInputReader(ABC):
+    @abstractmethod
+    def load_context(self) -> ExternalExecutionContextData:
+        raise NotImplementedError()
+
+
+class ExternalExecutionOutputWriter(ABC):
+    @abstractmethod
+    def send_notification(self, notification: Notification) -> None:
+        ...
+
+
+def bootstrap_input_reader() -> ExternalExecutionInputReader:
+    spec = _load_io_spec_from_env("input_reader")
+    mod = importlib.import_module(spec.get("module") or "dagster_externals.io.default")
+    cls = getattr(mod, spec.get("attribute") or "FileInputReader")
+    assert issubclass(cls, ExternalExecutionInputReader)
+    return cls(**spec.get("params", {}))
```

Comment:
why are we dynamically importing a library here? not sure we need to invent a plugin system